pinot-query-planner/src/main/java/org/apache/pinot/calcite/rel/rules/PinotImplicitTableHintRule.java

```diff
@@ -0,0 +1,283 @@
+      LOGGER.debug("Override implicit table hint for {} with explicit partition function: {}",
+          tableScan, partitionFunction);
+      return partitionFunction;
+    } else {
+      return tablePartitionInfo.getPartitionFunctionName();
+    }
+  }
+
+  /**
+   * Get the partition parallelism from the hint, if any, otherwise use the default value of 1.
+   */
+  private static int getPartitionParallelism(LogicalTableScan tableScan, Map<String, String> kvOptions) {
+    int partitionParallelism = 1;
+    String partitionParallelismStr = kvOptions.get(PinotHintOptions.TableHintOptions.PARTITION_PARALLELISM);
+    if (partitionParallelismStr != null) {
+      try {
+        int explicitPartitionParallelism = Integer.parseInt(partitionParallelismStr);
+        if (explicitPartitionParallelism != partitionParallelism) {
+          LOGGER.debug("Override implicit table hint for {} with explicit partition parallelism: {}", tableScan,
+              explicitPartitionParallelism);
+          partitionParallelism = explicitPartitionParallelism;
+        }
+      } catch (NumberFormatException e) {
+        throw new IllegalArgumentException(
+            "Invalid partition parallelism: " + partitionParallelismStr + " for table: " + tableScan);
+      }
+    }
+    return partitionParallelism;
+  }
+
+  /**
+   * Get the partition size from the hint, if any, otherwise use the partition size from the table partition info.
+   */
+  private static int getPartitionSize(TablePartitionInfo tablePartitionInfo, LogicalTableScan tableScan,
+      Map<String, String> kvOptions) {
+    int partitionSize = tablePartitionInfo.getNumPartitions();
+    String partitionSizeStr = kvOptions.get(PinotHintOptions.TableHintOptions.PARTITION_SIZE);
+    if (partitionSizeStr != null) {
+      try {
+        int explicitPartitionSize = Integer.parseInt(partitionSizeStr);
+        if (explicitPartitionSize != partitionSize) {
+          LOGGER.debug("Override implicit table hint for {} with explicit partition size: {}", tableScan,
+              explicitPartitionSize);
+          partitionSize = explicitPartitionSize;
+        }
+      } catch (NumberFormatException e) {
+        throw new IllegalArgumentException("Invalid partition size: " + partitionSizeStr + " for table: " + tableScan);
+      }
+    }
+    return partitionSize;
+  }
+
+  /**
+   * Get the table partition info for the given table scan, transforming Calcite's table name to Pinot's table name
+   * and adding the table type if needed.
+   */
+  @Nullable
+  private TablePartitionInfo getTablePartitionInfo(LogicalTableScan tableScan) {
+    List<String> qualifiedName = tableScan.getTable().getQualifiedName();
+    String tableName;
+    if (qualifiedName.size() == 1) {
+      tableName = qualifiedName.get(0);
+    } else {
+      tableName = DatabaseUtils.translateTableName(qualifiedName.get(1), qualifiedName.get(0));
+    }
```

Comment:
Use `RelToPlanNodeConverter.getTableNameFromTableScan(tableScan)`

Reply:
Is the change not applied yet?

Reply:
Yep, it has been changed. The code now is:

```java
 @Nullable
  private TablePartitionInfo getTablePartitionInfo(LogicalTableScan tableScan) {
    String tableName = RelToPlanNodeConverter.getTableNameFromTableScan(tableScan);
...
}
```